### PR TITLE
feat: integrate annlite with projector

### PR DESCRIPTION
**Pull request by jemmyshin**
This pr related to this [issue](https://github.com/jina-ai/annlite/issues/134)

Implement projector with `annlite`.

### CODE REVIEW

File: annlite/core/codec/projector.py [hunk 1]
```diff
@@ -31,13 +31,15 @@ class ProjectorCodec(BaseCodec):
...
         self,
+        dim: int,
         n_components: int = 128,
         whiten: Optional[bool] = False,
         svd_solver: Optional[str] = 'auto',
         is_incremental: Optional[bool] = False,
         batch_size: Optional[int] = 512,
     ):
         super(ProjectorCodec, self).__init__(require_train=True)
+        self.dim = dim
```
Comment (numb3r3) [inline annlite/core/codec/projector.py:42]: add `assert dim > n_components`

File: annlite/core/codec/projector.py [hunk 2]
```diff
@@ -66,6 +68,7 @@ def fit(self, x: 'np.ndarray'):
         """
         assert x.dtype == np.float32
         assert x.ndim == 2
+        assert x.shape[1] == self.dim, 'dimension of input data must be equal to "dim"'
```
Comment (numb3r3) [inline annlite/core/codec/projector.py:71]:
```suggestion
        assert x.shape == (2, self.dim), 'dimension of input data must be equal to "dim"'
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: annlite/core/codec/projector.py (hunk 2 above)
Comment (jemmyshin) [reply annlite/core/codec/projector.py]: why `x.shape[0] == 2` ?

Comment (numb3r3) [reply annlite/core/codec/projector.py]: oh, my fault.

File: annlite/index.py [hunk 3]
```diff
@@ -113,7 +129,9 @@ def __init__(
 
         super(AnnLite, self).__init__(
             dim=dim,
+            n_components=n_components,
```
Comment (numb3r3) [inline annlite/index.py:132]: we can get the `n_components` from self.project_codec

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: annlite/container.py [hunk 4]
```diff
@@ -33,14 +36,15 @@ def __init__(
         **kwargs,
     ):
         self.dim = dim
+        self.n_components = n_components
```
Comment (numb3r3) [inline annlite/container.py:39]:
```suggestion
        self.n_components = self.projector_codec.n_component if self.projector_codec else 0
```

File: annlite/container.py [hunk 5]
```diff
@@ -20,7 +21,9 @@ class CellContainer:
     def __init__(
         self,
         dim: int,
+        n_components: Optional[int] = None,
```
Comment (numb3r3) [inline annlite/container.py:24]: don't need pass `n_components` parameter in container __init__

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Review by numb3r3 — APPROVED** [review]

**Pull request merged**